cmd/broker/ingress/main.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"log"
+
+	"github.com/google/knative-gcp/pkg/broker/ingress"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"knative.dev/pkg/logging"
+)
+
+func main() {
+	logCfg := zap.NewProductionConfig() // TODO: to replace with a dynamically updating logger.
+	logCfg.EncoderConfig.EncodeTime = zapcore.ISO8601TimeEncoder
+	logger, err := logCfg.Build()
+	if err != nil {
+		log.Fatalf("Unable to create logger: %v", err)
+	}
+	ctx := context.Background()
+	logging.WithLogger(ctx, logger.Sugar())
+
+	ingress, err := ingress.NewHandler(ctx)
+	if err != nil {
+		logger.Fatal("Unable to create ingress handler: ", zap.Error(err))
+	}
+
+	logger.Info("Starting ingress.", zap.Any("ingress", ingress))
+	if err := ingress.Start(ctx); err != nil {
+		logger.Fatal("failed to start ingress: ", zap.Error(err))
+	}
```

Comment:
I believe it needs to handle term signal for graceful shutdown. So either add a TODO or just do it (we have example everywhere).

Reply:
Can you share an example? I am not sure what else needs to be handled here

Reply:
Sorry not familiar with this code base.  I tried to look for examples but didn't find one

Reply:
I think you can simply use `signals.NewContext()`, e.g. https://github.com/knative/eventing/blob/master/cmd/broker/ingress/main.go#L86

Reply:
Done. Thanks.